web-ui/src/main/resources/catalog/components/common/map/mapService.js

```diff
@@ -1078,24 +1078,31 @@
                   var vectorFormat = new ol.format.WFS();
               }
 
+
+              //check if proxy is needed
+              var _url = url.split('/');
+              _url = _url[0] + '/' + _url[1] + '/' + _url[2] + '/';
+              if ($.inArray(_url, gnGlobalSettings.requireProxy) >= 0
+                && url.indexOf(gnGlobalSettings.proxyUrl) != 0) {
+                getCapLayer.useProxy = true;
+              }
+              
+              var urlService = url;
+              
               var vectorSource = new ol.source.Vector({
                 format: vectorFormat,
-                loader: function(extent, resolution, projection) {
-                  if (this.loadingLayer) {
-                    return;
-                  }
-
-                  this.loadingLayer = true;
+                url: function(extent, resolution, projection) {
 
-                  var parts = url.split('?');
+                  var parts = urlService.split('?');
```

Comment:
Instead of doing the full ajax call, tell OL what is the URL where the features are and let OL do the work internally. They know better than us.